twoliter/src/lock/image.rs

```diff
@@ -0,0 +1,434 @@
+                    uri: ImageUri {
+                        registry: Some(vendor.registry.clone()),
+                        repo: image.name.to_string(),
+                        tag: format!("v{}", image.version),
+                    },
+                })
+            },
+        }
+    }
+
+    pub(crate) fn from_locked_image(
+        locked_image: &LockedImage,
+        vendor: &Vendor,
+        override_: Option<&Override>,
+    ) -> Self {
+        Self {
+            image_resolver_impl: if let Some(override_) = override_ {
+                Box::new(OverriddenImage {
+                    base_uri: ImageUri {
+                        registry: Some(vendor.registry.clone()),
+                        repo: locked_image.name.to_string(),
+                        tag: format!("v{}", locked_image.version),
+                    },
+                    override_: override_.clone(),
+                })
+            } else {
+                Box::new(VerbatimImage {
+                    uri: ImageUri {
+                        registry: Some(vendor.registry.clone()),
+                        repo: locked_image.name.to_string(),
+                        tag: format!("v{}", locked_image.version),
+                    },
+                })
+            },
+        }
+    }
+
+    /// Calculate the digest of the locked image
+    async fn calculate_digest(&self, image_tool: &ImageTool) -> Result<String> {
+        let image_uri = self.image_resolver_impl.uri();
+        let image_uri_str = image_uri.to_string();
+        let manifest_bytes = image_tool.get_manifest(image_uri_str.as_str()).await?;
+        let digest = sha2::Sha256::digest(manifest_bytes.as_slice());
+        let digest = base64::engine::general_purpose::STANDARD.encode(digest.as_slice());
+        trace!(
+            "Calculated digest for locked image '{}': '{}'",
+            image_uri,
+            digest,
+        );
+        Ok(digest)
+    }
+
+    async fn get_manifest(&self, image_tool: &ImageTool) -> Result<ManifestListView> {
+        let uri = self.image_resolver_impl.uri().to_string();
+        let manifest_bytes = image_tool.get_manifest(uri.as_str()).await?;
+        serde_json::from_slice(manifest_bytes.as_slice())
+            .context("failed to deserialize manifest list")
+    }
+
+    pub(crate) async fn resolve(
+        &self,
+        vendor: String,
+        image_tool: &ImageTool,
+        skip_metadata: bool,
+    ) -> Result<(LockedImage, Option<ImageMetadata>)> {
+        // First get the manifest list
+        let uri = self.image_resolver_impl.uri();
+        let manifest_list = self.get_manifest(image_tool).await?;
+        let registry = uri
+            .registry
+            .as_ref()
+            .context("no registry found for image")?;
+
+        let locked_image = LockedImage {
+            name: self.image_resolver_impl.name(),
+            version: self.image_resolver_impl.version()?,
+            vendor: vendor.to_string(),
```

Comment:
Do you think it makes sense for the `image_resolver_impl` to also be able to tell us the vendor?